src/services/expenses.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+let expenses = [];
+
+function getAllExpenses(userId, from, to, category) {
+  const filtredExpenses = expenses.filter(expense => (
+    expense.userId === Number(userId)
+    && category.includes(expense.category)
+    && Date.parse(expense.spentAt) > Date.parse(from)
+    && Date.parse(expense.spentAt) < Date.parse(to)
+  ));
+
+  return filtredExpenses;
+}
+
+function deleteExpense(expenseId) {
+  const filtredExpenses = expenses.filter(expense => (
+    expense.id !== Number(expenseId)
+  ));
+
+  if (expenses.length === filtredExpenses.length) {
+    return false;
+  }
+
+  expenses = filtredExpenses;
+
+  return true;
+}
+
+function getExpense(expenseId) {
+  return expenses.find(
+    expense => expense.id === Number(expenseId)
+  );
+}
+
+function createNewExpense(userId, spentAt, title, amount, category, note) {
+  let newId = 0;
+
+  if (expenses.length) {
+    newId = [...expenses].sort(
+      (expenseA, expenseB) => expenseB.id - expenseA.id
+    )[0].id + 1;
+  }
```

Comment:
this code is difficult to understand, don't try to write code in a couple of lines, create additional variables